crates/verkle/db/src/verkle_db.rs

```diff
@@ -0,0 +1,320 @@
+use reth_db::{
+    tables, transaction::DbTxMut
+};
+use reth_db_api::{
+    cursor::DbCursorRO,
+    transaction::DbTx,
+};
+use std::collections::HashMap;
+use verkle_trie::{from_to_bytes::{FromBytes, ToBytes}, database::{memory_db::MemoryDb, meta::{BranchChild, BranchMeta, StemMeta}, Flush, ReadOnlyHigherDb, WriteOnlyHigherDb}};
+
+// All nodes at this level or above will be cached in memory
+const CACHE_DEPTH: u8 = 4;
+pub(crate) const LEAF_TABLE_MARKER: u8 = 0;
+pub(crate) const STEM_TABLE_MARKER: u8 = 1;
+pub(crate) const BRANCH_TABLE_MARKER: u8 = 2;
+
+/// Database for operations over verkle trie.
+#[derive(Debug)]
+pub struct VerkleDb<'a, TX> {
+    /// The underlying key value database
+    // We try to avoid fetching from this, and we only store at the end of a batch insert
+    pub tx: &'a TX,
+    /// This stores the key-value pairs that we need to insert into the storage
+    // This is flushed after every batch insert
+    pub batch: MemoryDb,
+    /// This stores the top 3 layers of the trie, since these are the most accessed
+    // in the trie on average
+    pub cache: MemoryDb,
+}
```

Comment:
`VerkleDb` will the main db struct responsible for managing trie-operations, and finally flushing them to disk once cache-depth exceedes

Reply:
there are two approaches suggested in rust-verkle:

1.  see [rocksDB impl](https://github.com/crate-crypto/rust-verkle/blob/master/verkle-db/src/rocksdb_impl.rs) either we implement similar functionality for our MDBX or
2.  what I have done till now, create a DB struct and implement required traits for it
see [here](https://github.com/crate-crypto/rust-verkle/blob/ec29458dbe6a1cf6f77a02e367a29ea59f3ba3c2/verkle-trie/src/database.rs#L10) for more reference